### PR TITLE
User list for moderative actions

### PR DESCRIPTION
**Pull request by ateufel**
This PR is for a new component basically for moderators and supervisors. Supervisors can add/remove the moderator role to a user/contributor, moderators can ban users.

**Changelog**
* New menu entry in the sidebar for the list of users (contributors, moderators, supervisors).
* User list component with backend search.
* Single entries have tags for their user roles, if moderator or supervisor. No tag for simple contributors.
* There is a special "banned" tag for banned users, with a PopOver showing the ban details.
* Depending on the user role, several buttons for moderative actions show up (see screenshots).
* Banning users requires a reason and an end date, both data can be entered in a modal (see screenshots).

**Screenshots**

Tags/Buttons:
userlist_tags

Ban Modal:
userlist_ban_modal

Ban Popover:
userlist_ban_popover

### CODE REVIEW

File: src/components/Sidebar/index.js [hunk 1]
```diff
@@ -20,6 +20,7 @@ const CustomSidebar = ({location, user, articles}) => {
       <Menu theme="dark" mode="inline" defaultSelectedKeys={[location.pathname]} style={{height: '100%', borderRight: 0, marginTop: '20px'}}>
         {user.username && user.isContributor && <Menu.Item key="/mycontributions"><Link to="/mycontributions">My Contributions</Link></Menu.Item>}
         {/*{user.isModerator && */}<Menu.Item key="/review"><Link to="/review">Review</Link></Menu.Item>{/*}*/}
+        {/*{user.isSupervisor && */}<Menu.Item key="/users"><Link to="/users">Users</Link></Menu.Item>{/*}*/}
```
Comment (the-codingninja) [inline src/components/Sidebar/index.js]: We can uncomment this for production @ateufel

**Review by the-codingninja — COMMENTED** [review]
How can we unban right now?

File: src/components/Sidebar/index.js (hunk 1 above)
Comment (ateufel) [reply src/components/Sidebar/index.js]: Yeah, i will add it with the production flag, give me a minute!

**Comment by ateufel** [comment]
There is no backend API for unban at the moment, according to the backend docs. But it would be easy to implement if it is added.

**Comment by ateufel** [comment]
Unban function added in the frontend. Backend needs to add the unban function: `v1/moderation/unban`

**Pull request merged**